The Global Assembly Cache is not supported for net5.0 target

I changed [gitfool/BoardGameGeek.Dungeon](https://github.com/gitfool/BoardGameGeek.Dungeon) to [retarget .NET 5.0](https://github.com/gitfool/BoardGameGeek.Dungeon/commit/40eca7ef8ae20a6b377a489749ca5ad3d522976c) (RC1) and this flushed out an issue with `PocketContainer`, which I use indirectly via [`PocketLogger`](https://github.com/gitfool/BoardGameGeek.Dungeon/blob/master/Application/Application.csproj#L15). I pushed the build to GitHub anyway, to show the [failure](https://github.com/gitfool/BoardGameGeek.Dungeon/runs/1137056472#step:6:249):
```
/__w/BoardGameGeek.Dungeon/BoardGameGeek.Dungeon/.nuget/packages/pocket.typediscovery/0.4.0/contentFiles/cs/netstandard2.0/(Pocket)/Discover.cs(60,35): error CS0618: 'Assembly.GlobalAssemblyCache' is obsolete: 'The Global Assembly Cache is not supported.' [/__w/BoardGameGeek.Dungeon/BoardGameGeek.Dungeon/Application/Application.csproj]
```
Coming from: https://github.com/jonsequitur/PocketContainer/blob/6e5f8d8450a0d15d9d72632ad38f6e085c9013e1/Pocket.Container/Discover.cs#L56-L61
Use of `GlobalAssemblyCache` should probably be removed or wrapped by a conditional `#if` directive.

--- a/Pocket.Container.CumulativeRegistration/Pocket.Container.CumulativeRegistration.csproj
+++ b/Pocket.Container.CumulativeRegistration/Pocket.Container.CumulativeRegistration.csproj
@@ -1,11 +1,11 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
-  <PropertyGroup>    
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
-    <DefineConstants>TRACE;DEBUG;NETCOREAPP2_0;SourceProject</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;NETCOREAPP3_1;SourceProject</DefineConstants>
   </PropertyGroup>
 
   <ItemGroup>

--- a/Pocket.Container.CumulativeRegistration/Pocket.Container.CumulativeRegistration.csproj
+++ b/Pocket.Container.CumulativeRegistration/Pocket.Container.CumulativeRegistration.csproj
@@ -19,10 +19,10 @@
   </ItemGroup>
 
   <ItemGroup>
-  	<PackageReference Include="fluentassertions" Version="4.19.4" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
-    <PackageReference Include="xunit" Version="2.3.0-beta5-build3769" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.0-beta5-build3769" />
+  	<PackageReference Include="fluentassertions" Version="5.10.3" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.7.1" />
+    <PackageReference Include="xunit" Version="2.4.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3" />
   </ItemGroup>
 
 

--- a/Pocket.Container.CumulativeRegistration/PocketContainerCumulativeRegistrationTests.cs
+++ b/Pocket.Container.CumulativeRegistration/PocketContainerCumulativeRegistrationTests.cs
@@ -43,7 +43,7 @@ namespace Pocket.Container.CumulativeRegistration.Tests
 
             Action resolveEnumerable = () => container.Resolve<IEnumerable<int>>();
 
-            resolveEnumerable.ShouldThrow<ArgumentException>();
+            resolveEnumerable.Should().Throw<ArgumentException>();
         }
     }
 }

--- a/Pocket.Container.For.Microsoft.Extensions.DependencyInjection/Pocket.Container.For.Microsoft.Extensions.DependencyInjection.csproj
+++ b/Pocket.Container.For.Microsoft.Extensions.DependencyInjection/Pocket.Container.For.Microsoft.Extensions.DependencyInjection.csproj
@@ -1,13 +1,13 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
-  <PropertyGroup>    
-    <TargetFramework>netcoreapp2.0</TargetFramework>    
-    <AssemblyName>Pocket.Container.For.Microsoft.Extensions.DependencyInjection</AssemblyName>    
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <AssemblyName>Pocket.Container.For.Microsoft.Extensions.DependencyInjection</AssemblyName>
     <RootNamespace>Pocket.Container.For.Microsoft.Extensions.DependencyInjection</RootNamespace>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
-    <DefineConstants>TRACE;DEBUG;NETCOREAPP2_0;SourceProject</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;NETCOREAPP3_1;SourceProject</DefineConstants>
   </PropertyGroup>
 
   <ItemGroup>

--- a/Pocket.Container.For.Microsoft.Extensions.DependencyInjection/Pocket.Container.For.Microsoft.Extensions.DependencyInjection.csproj
+++ b/Pocket.Container.For.Microsoft.Extensions.DependencyInjection/Pocket.Container.For.Microsoft.Extensions.DependencyInjection.csproj
@@ -24,12 +24,12 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="fluentassertions" Version="4.19.4" />
-    <PackageReference Include="microsoft.extensions.dependencyinjection" Version="2.0.0" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
-    <PackageReference Include="pocket.disposable" Version="1.0.3" />
-    <PackageReference Include="xunit" Version="2.3.0-beta5-build3769" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.0-beta5-build3769" />
+    <PackageReference Include="fluentassertions" Version="5.10.3" />
+    <PackageReference Include="microsoft.extensions.dependencyinjection" Version="3.1.8" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.7.1" />
+    <PackageReference Include="pocket.disposable" Version="1.0.5" />
+    <PackageReference Include="xunit" Version="2.4.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3" />
   </ItemGroup>
 
   <ItemGroup>

--- a/Pocket.Container.For.Microsoft.Extensions.DependencyInjection/Tests/PocketContainerMicrosoftDependencyInjectionTests.cs
+++ b/Pocket.Container.For.Microsoft.Extensions.DependencyInjection/Tests/PocketContainerMicrosoftDependencyInjectionTests.cs
@@ -42,7 +42,7 @@ namespace Pocket.Container.For.Microsoft.Extensions.DependencyInjection.Tests
 
             Action resolve = () => container.GetRequiredService(typeof(string));
 
-            resolve.ShouldThrow<ArgumentException>();
+            resolve.Should().Throw<ArgumentException>();
         }
 
         [Fact]

--- a/Pocket.Container.Tests/Pocket.Container.Tests.csproj
+++ b/Pocket.Container.Tests/Pocket.Container.Tests.csproj
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
   </PropertyGroup>
   <ItemGroup>
     <Compile Remove="TestResults\**" />

--- a/Pocket.Container.Tests/Pocket.Container.Tests.csproj
+++ b/Pocket.Container.Tests/Pocket.Container.Tests.csproj
@@ -15,10 +15,10 @@
     <Compile Remove="Properties\AssemblyInfo.cs" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="FluentAssertions" Version="4.19.4" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
-    <PackageReference Include="xunit" Version="2.3.0-beta5-build3769" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.0-beta5-build3769" />
+    <PackageReference Include="FluentAssertions" Version="5.10.3" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.7.1" />
+    <PackageReference Include="xunit" Version="2.4.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\Pocket.Container\Pocket.Container.csproj" />

--- a/Pocket.Container.Tests/PocketContainerCloneTests.cs
+++ b/Pocket.Container.Tests/PocketContainerCloneTests.cs
@@ -125,7 +125,7 @@ namespace Pocket.Container.Tests
             Action resolveFromOriginal = () =>
                 original.Resolve<IEnumerable<string>>();
 
-            resolveFromOriginal.ShouldThrow<ArgumentException>();
+            resolveFromOriginal.Should().Throw<ArgumentException>();
         }
 
         [Fact]

--- a/Pocket.Container.Tests/PocketContainerCreateOverrideContainerTests.cs
+++ b/Pocket.Container.Tests/PocketContainerCreateOverrideContainerTests.cs
@@ -139,7 +139,7 @@ namespace Pocket.Container.Tests
 
             Action resolve = () => primary.Resolve<IEnumerable<string>>();
 
-            resolve.ShouldThrow<ArgumentException>();
+            resolve.Should().Throw<ArgumentException>();
         }
     }
 }

--- a/Pocket.Container.Tests/PocketContainerOpenGenericStrategyTests.cs
+++ b/Pocket.Container.Tests/PocketContainerOpenGenericStrategyTests.cs
@@ -52,7 +52,7 @@ namespace Pocket.Container.Tests
             Action registerWrongType = () =>
                 container.RegisterGeneric(typeof(string), typeof(List<>));
 
-            registerWrongType.ShouldThrow<ArgumentException>()
+            registerWrongType.Should().Throw<ArgumentException>()
                              .And
                              .Message
                              .Should()
@@ -67,7 +67,7 @@ namespace Pocket.Container.Tests
             Action registerWrongType = () =>
                 container.RegisterGeneric(typeof(IEnumerable<>), typeof(string));
 
-            registerWrongType.ShouldThrow<ArgumentException>()
+            registerWrongType.Should().Throw<ArgumentException>()
                              .And
                              .Message
                              .Should()

--- a/Pocket.Container.Tests/PocketContainerSingleImplementationStrategyTests.cs
+++ b/Pocket.Container.Tests/PocketContainerSingleImplementationStrategyTests.cs
@@ -51,7 +51,7 @@ namespace Pocket.Container.Tests
                 .IfOnlyOneImplementationUseIt();
 
             Action resolve = () => container.Resolve<IEnumerable>();
-            resolve.ShouldThrow<ArgumentException>();
+            resolve.Should().Throw<ArgumentException>();
         }
 
         public interface IHaveOnlyOneImplementation

--- a/Pocket.Container.Tests/PocketContainerTests.cs
+++ b/Pocket.Container.Tests/PocketContainerTests.cs
@@ -112,7 +112,7 @@ namespace Pocket.Container.Tests
 
             Action resolve = () => container.Resolve<SomeDelegateType>();
 
-            resolve.ShouldThrow<ArgumentException>()
+            resolve.Should().Throw<ArgumentException>()
                    .And
                    .Message
                    .Should()
@@ -162,7 +162,7 @@ namespace Pocket.Container.Tests
 
             Action resolveNotOptional = () => container.Resolve<HasOneParamCtor<IAmAnInterface>>();
 
-            resolveNotOptional.ShouldThrow<ArgumentException>();
+            resolveNotOptional.Should().Throw<ArgumentException>();
         }
 
         [Fact]
@@ -244,7 +244,7 @@ namespace Pocket.Container.Tests
             Action resolve = () =>
                 container.Resolve<HasTwoCtorsWithTheSameNumberOfParams>();
 
-            resolve.ShouldThrow<ArgumentException>()
+            resolve.Should().Throw<ArgumentException>()
                    .And
                    .Message
                    .Should()
@@ -259,7 +259,7 @@ namespace Pocket.Container.Tests
             Action resolve = () =>
                 container.Resolve<IAmAnInterface>();
 
-            resolve.ShouldThrow<ArgumentException>()
+            resolve.Should().Throw<ArgumentException>()
                    .And
                    .Message
                    .Should()
@@ -588,7 +588,7 @@ namespace Pocket.Container.Tests
 
             Action resolve = () => container.Resolve<IEnumerable<string>>();
 
-            resolve.ShouldThrow<DataMisalignedException>();
+            resolve.Should().Throw<DataMisalignedException>();
         }
 
         [Fact]
@@ -601,7 +601,7 @@ namespace Pocket.Container.Tests
 
             Action resolve = () => container.Resolve(typeof(IEnumerable<string>));
 
-            resolve.ShouldThrow<DataMisalignedException>();
+            resolve.Should().Throw<DataMisalignedException>();
         }
 
         [Fact]

--- a/Pocket.Container/Discover.cs
+++ b/Pocket.Container/Discover.cs
@@ -57,7 +57,9 @@ namespace Pocket
             AppDomain.CurrentDomain
                      .GetAssemblies()
                      .Where(a => !a.IsDynamic)
+#if NETFRAMEWORK
                      .Where(a => !a.GlobalAssemblyCache)
+#endif
                      .Types();
 
         /// <summary>

--- a/Pocket.Container/Pocket.Container.csproj
+++ b/Pocket.Container/Pocket.Container.csproj
@@ -1,11 +1,11 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
     <AssemblyName>PocketContainer</AssemblyName>
     <RootNamespace>PocketContainer</RootNamespace>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
-    <DefineConstants>TRACE;DEBUG;NETCOREAPP2_0;SourceProject</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;NETCOREAPP3_1;SourceProject</DefineConstants>
   </PropertyGroup>
   <ItemGroup>
     <Compile Remove="HttpConfigurationExtensions.cs" />

--- a/Pocket.TypeDiscovery.Tests/Pocket.TypeDiscovery.Tests.csproj
+++ b/Pocket.TypeDiscovery.Tests/Pocket.TypeDiscovery.Tests.csproj
@@ -1,11 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
-    <DefineConstants>TRACE;DEBUG;NETCOREAPP2_0;SourceProject</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;NETCOREAPP3_1;SourceProject</DefineConstants>
   </PropertyGroup>
 
   <ItemGroup>

--- a/Pocket.TypeDiscovery.Tests/Pocket.TypeDiscovery.Tests.csproj
+++ b/Pocket.TypeDiscovery.Tests/Pocket.TypeDiscovery.Tests.csproj
@@ -19,10 +19,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="fluentassertions" Version="4.19.4" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
-    <PackageReference Include="xunit" Version="2.3.0-beta5-build3769" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.0-beta5-build3769" />
+    <PackageReference Include="fluentassertions" Version="5.10.3" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.7.1" />
+    <PackageReference Include="xunit" Version="2.4.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3" />
   </ItemGroup>
 
 </Project>

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,12 +1,12 @@
 install:
-# install .NET CLI 2.0
-  - ps: $urlCurrent = "https://dotnetcli.azureedge.net/dotnet/Sdk/2.0.0/dotnet-sdk-2.0.0-win-x64.zip"
+# install .NET CLI
+  - ps: $urlCurrent = "https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.402/dotnet-sdk-3.1.402-win-x64.zip"
   - ps: $env:DOTNET_INSTALL_DIR = "$pwd\.dotnetsdk"
   - ps: mkdir $env:DOTNET_INSTALL_DIR -Force | Out-Null
   - ps: $tempFileCurrent = [System.IO.Path]::GetTempFileName()
   - ps: (New-Object System.Net.WebClient).DownloadFile($urlCurrent, $tempFileCurrent)
   - ps: Add-Type -AssemblyName System.IO.Compression.FileSystem; [System.IO.Compression.ZipFile]::ExtractToDirectory($tempFileCurrent, $env:DOTNET_INSTALL_DIR)
-  - ps: $env:Path = "$env:DOTNET_INSTALL_DIR;$env:Path"  
+  - ps: $env:Path = "$env:DOTNET_INSTALL_DIR;$env:Path"
   - ps: $env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
   - dotnet restore
 
@@ -18,12 +18,12 @@ build_script:
       dotnet build PocketContainer.sln /p:Version=%APPVEYOR_BUILD_VERSION%
 
 after_build:
-  - nuget pack Pocket.Container\PocketContainer.nuspec  
-  - nuget pack Pocket.Container\PocketContainerOpenGenericStrategy.nuspec  
-  - nuget pack Pocket.Container\PocketContainerPrimitiveAvoidanceStrategy.nuspec  
-  - nuget pack Pocket.Container\PocketContainer.AfterCreating.nuspec  
-  - nuget pack Pocket.Container\PocketContainer.Clone.nuspec  
-  - nuget pack Pocket.Container\PocketContainerSingleImplementationStrategy.nuspec  
+  - nuget pack Pocket.Container\PocketContainer.nuspec
+  - nuget pack Pocket.Container\PocketContainerOpenGenericStrategy.nuspec
+  - nuget pack Pocket.Container\PocketContainerPrimitiveAvoidanceStrategy.nuspec
+  - nuget pack Pocket.Container\PocketContainer.AfterCreating.nuspec
+  - nuget pack Pocket.Container\PocketContainer.Clone.nuspec
+  - nuget pack Pocket.Container\PocketContainerSingleImplementationStrategy.nuspec
   - nuget pack Pocket.Container\TypeDiscovery.nuspec
   - nuget pack Pocket.Container.CumulativeRegistration\PocketContainer.CumulativeRegistration.nuspec
   - nuget pack Pocket.Container.For.Microsoft.Extensions.DependencyInjection\PocketContainer.For.MicrosoftExtensionsDependencyInjection.nuspec

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,3 +1,5 @@
+image: Visual Studio 2019
+
 install:
 # install .NET CLI
   - ps: $urlCurrent = "https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.402/dotnet-sdk-3.1.402-win-x64.zip"

